HabitTracker.ConsoleApp-DreamFXX/Program.cs

```diff
@@ -0,0 +1,366 @@
+                    Id INTEGER PRIMARY KEY AUTOINCREMENT,
+                    Name TEXT NOT NULL,
+                    Unit TEXT NOT NULL
+                    )";
+            tableCmd.ExecuteNonQuery();
+
+            tableCmd.CommandText =
+                @"CREATE TABLE IF NOT EXISTS HabitRecords (
+                  Id INTEGER PRIMARY KEY AUTOINCREMENT,
+                  HabitId INTEGER,
+                  Date Text,
+                  Time Text,
+                  Quantity INTEGER,
+                  FOREIGN KEY (HabitId) REFERENCES Habits(Id)
+                  )";
+            tableCmd.ExecuteNonQuery();
+
+            connection.Close();
+
+        }
+
+        FillDatatables();
+        GetUserInput();
+    }
+
+    static void GetUserInput()
+    {
+        Console.Clear();
+
+        bool closeApp = false;
+        while (closeApp == false)
+        {
+            Console.WriteLine("Welcome to Habit Tracker!\n\n");
+            Console.WriteLine("MAIN MENU");
+            Console.WriteLine("0 -> Save and Exit App\n");
+            Console.WriteLine("------------------------------------------");
+            Console.WriteLine("1 -> Show All records.");
+            Console.WriteLine("2 -> Add a record.");
+            Console.WriteLine("3 -> Delete a record");
+            Console.WriteLine("4 -> Modify a record.");
+            Console.WriteLine("5 -> Add your own Habit to this App.");
+            Console.WriteLine("------------------------------------------");
+
+            string command = Console.ReadLine();
+
+            switch (command)
+            {
+                case "0":
+                    Console.WriteLine("\nHave a good day!\n");
+                    closeApp = true;
+                    Environment.Exit(1);
+                    break;
+                case "1":
+                    ViewAllRecords();
+                    break;
+                case "2":
+                    AddRecord();
+                    break;
+                case "3":
+                    DeleteRecord();
+                    break;
+                case "4":
+                    ChangeRecord();
+                    break;
+                case "5":
+                    AddNewHabit();
+                    break;
+                default:
+                    Console.WriteLine("\n\nInvalid number of operation. Try again, valid operations are in range 0 - 5.\n\n");
+                    break;
+            }
+        }
+    }
+
+    private static void ViewAllRecords()
+    {
+        Console.Clear();
+
+        using (var connection = new SQLiteConnection(connectionString))
+        {
+            connection.Open();
+            var tableCmd = connection.CreateCommand();
+            tableCmd.CommandText = "SELECT * FROM HabitRecords";
+
+            List<HabitRecord> tableData = new List<HabitRecord>();
+            SQLiteDataReader reader = tableCmd.ExecuteReader();
+
+            if (reader.HasRows)
```

Comment:
🟠 Code Complexity

💡 `if (reader.HasRows)` is a redundant level of indentation here, as `while (reader.Read())` will only be true if the reader has rows. Essentially, `reader.Read()` already does this check for you.

🔧 Remove the redundant `if` statement and only keep the `while` statement.